### PR TITLE
click==8.1.3

### PR DESCRIPTION
**Pull request by altendky**
Draft for:
- [x] Figuring out what to do with lgtm
- [x] Testing
- [x] Reviewing changelogs for 8+ https://click.palletsprojects.com/en/8.1.x/changes/
- [x] Special consideration of changes to wildcard handling on Windows.
  - When taking arguments from sys.argv on Windows, glob patterns, user dir, and env vars are expanded. [#1096](https://github.com/pallets/click/issues/1096)
  - Maybe we can punt on handling this: Pass `windows_expand_args=False` when calling the main command to disable pattern expansion on Windows. There is no way to escape patterns in CMD, so if the program needs to pass them on as-is then expansion must be disabled. [#1901](https://github.com/pallets/click/issues/1901)
- [x] `version_option` uses `importlib.metadata` (or the `importlib_metadata` backport) instead of `pkg_resources`. The version is detected based on the package name, not the entry point name. The Python package name must match the installed package name, or be passed with `package_name=`. [#1582](https://github.com/pallets/click/issues/1582)
- [x] A flag option with `required=True` requires that the flag is passed instead of choosing the implicit default value. [#1978](https://github.com/pallets/click/issues/1978)
- [x] Rely on [PEP 538](https://peps.python.org/pep-0538/) and [PEP 540](https://peps.python.org/pep-0540/) to handle selecting UTF-8 encoding instead of ASCII. Click’s locale encoding detection is removed. [#2198](https://github.com/pallets/click/issues/2198)
  - [x] Maybe we can drop `monkey_patch_click()`?
  https://github.com/Chia-Network/chia-blockchain/blob/a691d3c4b2aad546b8d9058b6a3882416edaee9b/chia/cmds/chia.py#L33-L46
  - Monkey-patch click instead of forking. https://github.com/Chia-Network/chia-blockchain/commit/be4af4128cd7900666c1028ba1545dfa06ed7391
  - https://github.com/pallets/click/pull/2199/files
  - https://peps.python.org/pep-0538/
  - https://peps.python.org/pep-0540/
- [x] Maybe drop types-click

### CODE REVIEW

**Review by arvidn — DISMISSED** [review]

File: chia/full_node/weight_proof.py [hunk 1]
```diff
@@ -724,7 +724,7 @@ def _get_weights_for_sampling(
     queries = -WeightProofHandler.LAMBDA_L * math.log(2, prob_of_adv_succeeding)
     for i in range(int(queries) + 1):
         u = rng.random()
-        q = 1 - delta ** u
+        q = 1 - delta**u
```
[the remark this answers is not in the document]
Comment (altendky) [reply chia/full_node/weight_proof.py:727]: Yes.

File: setup.py [hunk 2]
```diff
@@ -52,13 +51,11 @@
     "isort",
     "flake8",
     "mypy",
-    # TODO: black 22.1.0 requires click>=8, remove this pin after updating to click 8
```
[the remark this answers is not in the document]
Comment (altendky) [reply setup.py:56]: :+1:

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -46,10 +46,9 @@ repos:
     -   id: check-ast
     -   id: debug-statements
 -   repo: https://github.com/psf/black
-    rev: 21.12b0
+    rev: 22.3.0
```
[the remark this answers is not in the document]
Comment (altendky) [reply .pre-commit-config.yaml:49]: It is related in that new black requires new click and old black requires old click.

File: chia/util/click.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+) -> Callable[[F], _Command]:
+    # TODO: I believe that this can be resolved and is related to the order of the
+    #       overloads for click.command()
+    return click.command(name=name, cls=_Command, **attrs)  # type: ignore[return-value]
+
+
+def group(name: Optional[str] = None, **attrs: Any) -> Callable[[F], _Group]:
+    # TODO: I believe that this can be resolved and is related to the order of the
+    #       overloads for click.group()
+    return click.group(name=name, cls=_Group, **attrs)  # type: ignore[return-value]
```
[the remark this answers is not in the document]
Comment (altendky) [reply chia/util/click.py:71]: I did play with the CLI some, yes.  And yes, we should have plenty more CLI testing.

**Review by arvidn — DISMISSED** [review]

**Comment by arvidn** [comment]
you should probably fix the LGTM warnings too

**Comment by altendky** [comment]
probably getting replaced by https://github.com/Chia-Network/chia-blockchain/pull/13559

**Pull request closed without merging**